src/sim_engine.rs

```diff
@@ -38,7 +38,6 @@ impl Engine for SimEngine {
 
         let deleted_pool = self.pools.remove(name);
 
-        // TODO deal with not found
```

Comment:
Can you explain more? I lean towards returning a 'notfound' error, since I can think of cases where the client might want to know it passed a bad pool name.